app/models/cart.rb

```diff
@@ -38,18 +38,20 @@ def total
   end
 
   def add_quantity(quantity, item, inventory)
-    if @contents.values.first < inventory
+    if @contents[item.id.to_s] < inventory
       quantity.times do
         add_item(item.id.to_s)
       end
     end
   end
 
   def remove_quantity(quantity, item)
-    if @contents.values.first > 0
+    if @contents[item.id.to_s] > 1
       quantity.times do
         remove_item(item.id.to_s)
       end
+    else
+      return false
```

Comment:
if the `@contents[item.id.to_s] > 1` is below 1 it will return false which will make the controller delete the item instead of removing quantity.